docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md

```diff
@@ -14,305 +14,194 @@ weight: 700
-1. Query the `/emails` endpoint of the OAuth provider's API (configured with `api_url`), then check for the presence of an email address marked as a primary address.
-1. If no email address is found in steps (1-4), then the email address of the user is set to an empty string.
+Group mappings are made using [JMESPath](http://jmespath.org/examples.html) from the `groups_attribute_path` configuration option.
+Grafana determines a user's groups by following the steps below until it finds the groups:
 
-### Groups / Teams
+1. Evaluate the `groups_attribute_path` JMESPath expression against the OAuth2 ID token.
+1. Evaluate the `groups_attribute_path` JMESPath expression against the JSON data obtained from UserInfo endpoint, which is specified via the `api_url` configuration option.
 
-Similarly, group mappings are made using [JMESPath](http://jmespath.org/examples.html) with the `groups_attribute_path` configuration option. The `id_token` is attempted first, followed by the UserInfo from the `api_url`. The result of the JMESPath expression should be a string array of groups.
+The result of the JMESPath expression should be a string array of groups.
 
-Furthermore, Grafana will check for the presence of at least one of the teams specified via the `team_ids` configuration option using the [JMESPath](http://jmespath.org/examples.html) specified via the `team_ids_attribute_path` configuration option. The JSON used for the path lookup is the HTTP response obtained from querying the Teams endpoint specified via the `teams_url` configuration option (using `/teams` as a fallback endpoint). The result should be a string array of Grafana Team IDs. Using this setting ensures that only certain teams is allowed to authenticate to Grafana using your OAuth provider.
+You can limit access to Grafana to only members of a given group or list of groups by setting the `allowed_groups` option.
 
-You can limit access to only members of a given group or list of groups by setting the `allowed_groups` option.
+### Teams
 
-### Login
+OAuth2 provider teams are extracted using [JMESPath](http://jmespath.org/examples.html) expression from the `team_ids_attribute_path` configuration option.
+The expression is evaluated against the JSON data obtained from Teams endpoint, which is specified via the `teams_url` configuration option.
+The result should be a string array of OAuth2 provider team IDs.
 
-Customize user login using `login_attribute_path` configuration option. Order of operations is as follows:
+If `team_ids` configuration option is set, only users who are members of at least one of the specified teams will be able to authenticate to Grafana using your OAuth2 provider.
```

Comment:
This does seem relevant and important to mention. Can this bit be included in the table which means we can remove this section?